### ISSUE TITLE
Just a check about CNTK usage in Python

### ISSUE THREAD

**Issue by christopher5106** [Microsoft/CNTK#3607]
Hi,

I've written an example using CNTK operations, without high level packages
https://github.com/christopher5106/exercices/blob/master/cntk-ex.py
I have a few questions to be sure I'm using CNTK correctly:
1. I think it is necessary to write input shape=(-1, 2) with the batch dimension, right ? If I remove the batch dimension, the model does not converge any more.
2. I'm computing loss and gradients with 2 different functions, y.grad() and y.eval(). Is there a way to have both togeter ?

I've written the same example using the high level packages:
https://github.com/christopher5106/exercices/blob/master/cntk-ex_with_packages.py
1. I had to rebuild the sgd object since learning rate is decaying. How can we define a learning rate decay otherwise ?
2. Is it possible to do loss and gradient computations and parameter update ?

**Comment by delzac** [comment]
First, there's no need to define a static free axis for your batch dimension. Static free axis is your -1 in input shape=(-1, 2).

CNTK is different from other frameworks in that it has something called dynamic axis which is used to abstract away the batch dimension.

Long story short, you can define it as C.input_variable(shape=(2,)) and it has a dynamic batch axis already.

Also, is there any reasons that you are computing gradient and doing the updates manually? There are builtin cntk [trainers](https://www.cntk.ai/pythondocs/_modules/cntk/train/trainer.html#Trainer) for network training purposes.

To define a learning rate decay you can use [learning_parameter_schedule](https://cntk.ai/pythondocs/cntk.learners.html#cntk.learners.learning_parameter_schedule)

**Comment by christopher5106** [comment]
I'm sorry , in https://github.com/christopher5106/exercices/blob/master/cntk-ex.py when I remove the -1 in the shape, and modify functions, it does not converge anymore :( 
I do not see how to write it correctly.

**Comment by delzac** [comment]
I think its due to your cross_entropy function that you defined. axis=1 doesn't exist anymore.

**Comment by christopher5106** [comment]
I've updated my two examples as you adviced.
The problem was in the cross entropy function, which required an `unpack_batch` call to get a scalar for the loss of the batch.
